graders/c/Dockerfile

```diff
@@ -1,16 +1,17 @@
-FROM ubuntu:22.04
+FROM ubuntu:24.04
 
 LABEL maintainer="jonatan@yorku.ca"
 
 ARG DEBIAN_FRONTEND=noninteractive
 
-RUN apt-get update
-RUN apt-get install -y python3.10 gcc make python3-pip valgrind check pkg-config && apt clean
+RUN apt-get update && \
+    apt-get install -y python3.12 gcc make python3-pip valgrind check pkg-config && \
+    apt-get clean
 
-RUN ln -sf /usr/bin/python3.10 /usr/bin/python3
+RUN ln -sf /usr/bin/python3.12 /usr/bin/python3
```

Comment:
I think this is the biggest potential breaking change, e.g. if someone has binary wheels that were previously built against `3.10` that they included in the image via `serverFilesCourse`. I think this is unlikely, and unquestionably the right way for someone to do this is to build a custom image and layer in whatever stuff they need. Just food for thought.

Reply:
I believe that the chances of someone [relying on this in the C grader](https://xkcd.com/1172/) is pretty small, I'm willing to ignore this risk.

Reply:
I'm also willing to ignore it.